src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -57,65 +61,101 @@ var kubeconfigCmd = &cobra.Command{
 		cmd.SilenceUsage = true
 
 		configDir := path.Join(homedir.HomeDir(), ".kube")
-		linkPath := path.Join(configDir, "config")
-		if show {
-			// The output is "true", "false", or an error message for UI.
-			// We will only return nil in this path.
-			target, err := os.Readlink(linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					fmt.Println("false")
-				} else if errors.Is(err, syscall.EINVAL) {
-					fmt.Printf("File %s exists and is not a symlink\n", linkPath)
-				} else {
-					fmt.Printf("%s\n", err)
-				}
-			} else if target == configPath {
-				fmt.Println("true")
-			} else {
-				// For a symlink pointing elsewhere, we assume we can overwrite.
-				fmt.Println("false")
-			}
-			return nil
+
+		configFile, err := os.Open(configPath)
+		if err != nil {
+			return err
 		}
+
+		var config kubeConfig
+		defer configFile.Close()
+		err = yaml.NewDecoder(configFile).Decode(&config)
+		if err != nil {
+			return err
+		}
+
+		cleanConfig := removeExistingRDCluster(rdCluster, &config)
+
+		kubeConfig, err := updateClusterIP(*cleanConfig, rdNetworking)
+
+		var finalKubeConfigFile *os.File
 		if enable {
-			err = os.Mkdir(configDir, 0o750)
-			if err != nil && !errors.Is(err, os.ErrExist) {
-				// The error already contains the full path, we can't do better.
+			if err := os.MkdirAll(configDir, 0o750); err != nil {
 				return err
 			}
-			err = os.Symlink(configPath, linkPath)
+			finalKubeConfigFile, err = os.Create(filepath.Join(configDir, "config"))
```

Comment:
We used to not touch the file if it exists; we now truncate it. This could get awkward if they wanted the integration for the docker socket but not the kubeconfig munging…